### PR TITLE
Bugifx/fix query and saving

### PR DESCRIPTION
**Pull request by millerts**

### CODE REVIEW

File: src/main/java/org/opentestsystem/ap/imrt/common/model/BaseItem.java [hunk 1]
```diff
@@ -71,8 +72,7 @@
     @NotNull
     private org.opentestsystem.ap.common.model.Item itemJson;
 
-    @OneToMany
-    @JoinColumn(name = "item_key")
+    @OneToMany(mappedBy = "standardIdKey.item", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
```
Comment (millerts) [inline src/main/java/org/opentestsystem/ap/imrt/common/model/BaseItem.java]: So this is the magic that took me way too long to figure out. The standardIdKey.item is enough for JPA/Hibernate to know that you mean look at the StandardIdKey's item property within StandardId. Most of the examples around using OneToMany don't use composite keys.

Long term we may want to consider removing the BaseItem from the StandardIdKey and instead save standard ids with the item by using this field.  The reason being is we have a bidirectional and one could himself or herself in a circular loop.  And we don't fetch standard ids without their item.

File: src/test/java/org/opentestsystem/ap/imrt/common/repository/ImrtItemRepositoryTest.java [hunk 2]
```diff
@@ -119,4 +123,51 @@ public void shouldFindByStimId() {
...
         assertThat(result).contains(item1, item3);
     }
+
+    @Test
+    public void shouldUpdateItemsWithStandardId() {
+        ImrtItem imrtItem = new ImrtItemBuilder()
+                .withKey(null)
+                .build();
+        // After a successful save, the returned object should be equal to the original object
+        imrtItemRepository.save(imrtItem);
```
Comment (swilson11) [inline src/test/java/org/opentestsystem/ap/imrt/common/repository/ImrtItemRepositoryTest.java]: Did you mean to add a check on the returned object == original object?

File: src/main/java/org/opentestsystem/ap/imrt/common/model/BaseItem.java (hunk 1 above)
Comment (swilson11) [reply src/main/java/org/opentestsystem/ap/imrt/common/model/BaseItem.java]: I agree that would be better - it is a pretty easy change on the ingest side.

**Review by swilson11 — APPROVED** [review]

File: src/test/java/org/opentestsystem/ap/imrt/common/repository/ImrtItemRepositoryTest.java (hunk 2 above)
Comment (jeffjohnson9046) [reply src/test/java/org/opentestsystem/ap/imrt/common/repository/ImrtItemRepositoryTest.java]: I think this is just part of the setup for this test, making sure we have an existing item that we want to update.  As such, I dunno that we need to verify what we saved matches the original (my assumption is that's tested somewhere else).

this is in response to @swilson11 's comment.

**Review by jeffjohnson9046 — APPROVED** [review]

File: src/test/java/org/opentestsystem/ap/imrt/common/repository/ImrtItemRepositoryTest.java (hunk 2 above)
Comment (swilson11) [reply src/test/java/org/opentestsystem/ap/imrt/common/repository/ImrtItemRepositoryTest.java]: That's fine, then just remove the comment about them being equal

Comment (millerts) [reply src/test/java/org/opentestsystem/ap/imrt/common/repository/ImrtItemRepositoryTest.java]: Will do.  Sorry about that.  I modified this test a few ways to make sure we tested all the different ways to save the item.

**Pull request merged**